File: src/main/java/com/esprit/service/StudentService.java
package com.esprit.service;

import java.security.Principal;
import java.util.List;

import org.mapstruct.Named;

import com.esprit.dto.student.CreateStudentRequest;
import com.esprit.dto.student.StudentDTO;

public interface StudentService {

	void addStudent(CreateStudentRequest createStudentRequest);

	void addAllStudent(List<CreateStudentRequest> createStudentRequest);

	void updateStudent(CreateStudentRequest createStudentRequest);

	void deleteStudent(String studentId);

	StudentDTO findByPrincipal(Principal principal);

	List<StudentDTO> findAll();

	List<StudentDTO> findByClass(String classe);

	@Named("countByClass")
	int countByClass(String classe);
	
	int countByClassAndGroup(String classe, String group);

}

File: src/webapp/src/app/core/services/student-service.ts
import { HttpClient, HttpResponse } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';;
import { ConfigService } from '@ngx-config/core';
import { CreateStudentRequest, Student } from '../models';

@Injectable()
export class StudentService {

    readonly domain: string;

    constructor(private config: ConfigService,
        private httpClient: HttpClient) {
        this.domain = this.config.getSettings('environment.serviceBaseUrl') + "student/";
    }

    addStudent(data: CreateStudentRequest): Observable<any> {
        return this.httpClient.post<any>(this.domain, data, { observe: 'response' });
    }

    updateStudent(data: CreateStudentRequest): Observable<any> {
        return this.httpClient.put<any>(this.domain, data, { observe: 'response' });
    }

    addAllStudent(data: CreateStudentRequest[]): Observable<any> {
        return this.httpClient.post<any>(this.domain + "all", data, { observe: 'response' });
    }

    countByClassAndGroup(classe, group): Observable<any> {
        return this.httpClient.get<any>(this.domain + "class/" + classe + "/group/" + group);
    }

    getStudents(): Observable<any[]> {
        return this.httpClient.get<any[]>(this.domain);
    }

    getCurrentStudent(): Observable<any[]> {
        return this.httpClient.get<any[]>(this.domain + "current-student");
    }

    deleteStudent(id) {
        return this.httpClient.delete<void>(this.domain + id);
    }

}